File: internal/models/system.go
package models

type SystemStats struct {
	CPU       CPUStats    `json:"cpu"`
	Memory    MemoryStats `json:"memory"`
	Disk      DiskStats   `json:"disk"`
	Uptime    int64       `json:"uptimeSeconds"`
	LoadAvg   [3]float64  `json:"loadAvg"`
	Processes int         `json:"processes"`
}

type CPUStats struct {
	Percent float64 `json:"percent"`
	Cores   int     `json:"cores"`
}

type MemoryStats struct {
	TotalMB int64   `json:"totalMb"`
	UsedMB  int64   `json:"usedMb"`
	FreeMB  int64   `json:"freeMb"`
	Percent float64 `json:"percent"`
}

type DiskStats struct {
	TotalGB int64   `json:"totalGb"`
	UsedGB  int64   `json:"usedGb"`
	FreeGB  int64   `json:"freeGb"`
	Percent float64 `json:"percent"`
}

File: dashboard/src/interfaces/system.ts
export interface SystemStats {
  cpu: CPUStats;
  disk: DiskStats;
  memory: MemoryStats;
  loadAvg: number[];
  processes: number;
  uptimeSeconds: number;
  docker: DockerStats;
  backups?: { size: string };
  packageCache?: { size: string };
  systemLogs?: { size: string };
}

export interface DockerStats {
  images: DockerLayerStat;
  containers: DockerLayerStat;
  volumes: DockerLayerStat;
  buildCache: DockerLayerStat;
  reclaimableGb: number;
}

export interface DockerLayerStat {
  active: string;
  totalCount: string;
  size: string;
  reclaimable: string;
}

export interface CPUStats {
  cores: number;
  percent: number;
}

export interface DiskStats {
  freeGb: number;
  percent: number;
  totalGb: number;
  usedGb: number;
}

export interface MemoryStats {
  freeMb: number;
  percent: number;
  totalMb: number;
  usedMb: number;
}
